src/mame/namco/namcos10.cpp

```diff
@@ -3457,20 +3457,29 @@ static INPUT_PORTS_START( ippo2 )
 	PORT_INCLUDE(namcos10)
 
 	PORT_MODIFY("IN1")
-	PORT_BIT( 0x1ff33f61, IP_ACTIVE_LOW, IPT_UNUSED )
-
-	PORT_BIT( 0x00008000, IP_ACTIVE_LOW, IPT_SELECT )
+	PORT_BIT( 0x07ff0f61, IP_ACTIVE_LOW, IPT_UNUSED )
 
 	// TODO: These sensors must match what the game is expecting based on when it's trying to raise the punch pad or else it will throw a speed error
 	PORT_BIT( 0x00000002, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_NAME("Speed Sensor Down")
 	PORT_BIT( 0x00000004, IP_ACTIVE_HIGH, IPT_BUTTON2 ) PORT_NAME("Speed Sensor Mid")
 	PORT_BIT( 0x00000008, IP_ACTIVE_HIGH, IPT_BUTTON3 ) PORT_NAME("Speed Sensor Up")
 
-	PORT_BIT( 0x00040000, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_NAME("Human Sensor 1")
-	PORT_BIT( 0x00080000, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("Human Sensor 2")
-	PORT_BIT( 0x00004000, IP_ACTIVE_LOW, IPT_BUTTON6 ) PORT_NAME("Human Sensor 3")
-
+	// If the safety sensor is still triggered when the speed sensor is at the mid or up position then the game
+	// pauses and displays an message saying that something obstructing the safety sensor
 	PORT_BIT( 0x00000010, IP_ACTIVE_HIGH, IPT_BUTTON7 ) PORT_NAME("Safety Sensor")
+
+	// If none of the following sensors are set then the game pauses and tells the player to step closer to the machine
+	PORT_BIT( 0x00001000, IP_ACTIVE_HIGH, IPT_BUTTON4 ) PORT_NAME("Human Sensor 1")
+	PORT_BIT( 0x00002000, IP_ACTIVE_HIGH, IPT_BUTTON5 ) PORT_NAME("Human Sensor 2")
+	PORT_BIT( 0x00004000, IP_ACTIVE_HIGH, IPT_BUTTON6 ) PORT_NAME("Human Sensor 3")
+
+	PORT_BIT( 0x00008000, IP_ACTIVE_LOW, IPT_SELECT )
+
+	PORT_BIT( 0x08000000, IP_ACTIVE_LOW, IPT_TILT )
+
+	PORT_BIT( 0x10000000, IP_ACTIVE_LOW, IPT_COIN2 ) PORT_IMPULSE(1)
+	PORT_BIT( 0x20000000, IP_ACTIVE_LOW, IPT_COIN1 ) PORT_IMPULSE(1)
```

Comment:
This attempts to fix a coin error that would occasionally happened when you coined in.

Reply:
most Taito games have this whereby if you hold a coin for too long they'll throw a coin error.  A lot of the time we don't bother to limit it though (I think some secret test mode codes require the input to be held in certain cases)

Reply:
Well in this case it actually hinders things so it needed to be fixed to even continue working on it in my WIP branch.